src/bigml/api/core.clj

```diff
@@ -41,6 +41,13 @@
   `(binding [*connection* (merge *connection* {:dev-mode true})]
      ~@body))
 
+(defmacro with-mode
```

Comment:
It seems a bit repetitive to have both `with-mode` and `with-dev-mode`.  Perhaps this could replace the `with-dev-mode` defined above, so that `with-dev-mode` takes a boolean.